planner/core/testdata/enforce_mpp_suite_out.json

```diff
@@ -1203,5 +1203,201 @@
+      },
+      {
+        "SQL": "select count(distinct a), count(distinct b), sum(c) from t",
+        "Plan": [
+          "5 5 30000"
+        ],
+        "Warn": null
+      },
+      {
+        "SQL": "EXPLAIN select count(distinct a, b), count(distinct b), count(c), sum(d) from t",
+        "Plan": [
+          "TableReader_26 1.00 root  data:ExchangeSender_25",
+          "└─ExchangeSender_25 1.00 mpp[tiflash]  ExchangeType: PassThrough",
+          "  └─Projection_21 1.00 mpp[tiflash]  Column#6, Column#7, Column#8, Column#9",
+          "    └─HashAgg_22 1.00 mpp[tiflash]  funcs:count(distinct test.t.a, test.t.b)->Column#6, funcs:count(distinct test.t.b)->Column#7, funcs:sum(Column#12)->Column#8, funcs:sum(Column#13)->Column#9",
+          "      └─ExchangeReceiver_24 1.00 mpp[tiflash]  ",
+          "        └─ExchangeSender_23 1.00 mpp[tiflash]  ExchangeType: PassThrough",
+          "          └─HashAgg_20 1.00 mpp[tiflash]  group by:Column#16, Column#17, funcs:count(Column#14)->Column#12, funcs:sum(Column#15)->Column#13",
+          "            └─Projection_27 10000.00 mpp[tiflash]  test.t.c, cast(test.t.d, decimal(10,0) BINARY)->Column#15, test.t.a, test.t.b",
+          "              └─TableFullScan_11 10000.00 mpp[tiflash] table:t keep order:false, stats:pseudo"
+        ],
+        "Warn": [
+          "Some grouping sets should be merged",
+          "Some grouping sets should be merged"
+        ]
+      },
+      {
+        "SQL": "select count(distinct a, b), count(distinct b), count(c), sum(d) from t",
+        "Plan": [
+          "5 5 10 10"
+        ],
+        "Warn": [
+          "Some grouping sets should be merged",
+          "Some grouping sets should be merged"
+        ]
+      },
+      {
+        "SQL": "EXPLAIN select count(distinct a+b), sum(c) from t",
+        "Plan": [
+          "TableReader_26 1.00 root  data:ExchangeSender_25",
+          "└─ExchangeSender_25 1.00 mpp[tiflash]  ExchangeType: PassThrough",
+          "  └─Projection_21 1.00 mpp[tiflash]  Column#6, Column#7",
+          "    └─HashAgg_22 1.00 mpp[tiflash]  funcs:count(distinct Column#10)->Column#6, funcs:sum(Column#11)->Column#7",
+          "      └─ExchangeReceiver_24 1.00 mpp[tiflash]  ",
+          "        └─ExchangeSender_23 1.00 mpp[tiflash]  ExchangeType: PassThrough",
+          "          └─HashAgg_20 1.00 mpp[tiflash]  group by:Column#13, funcs:sum(Column#12)->Column#11",
+          "            └─Projection_27 10000.00 mpp[tiflash]  cast(test.t.c, decimal(10,0) BINARY)->Column#12, plus(test.t.a, test.t.b)->Column#13",
+          "              └─TableFullScan_11 10000.00 mpp[tiflash] table:t keep order:false, stats:pseudo"
+        ],
+        "Warn": null
+      },
+      {
+        "SQL": "select count(distinct a+b), sum(c) from t",
+        "Plan": [
+          "5 30000"
+        ],
+        "Warn": null
+      },
+      {
+        "SQL": "EXPLAIN select count(distinct a+b), count(distinct b+c), count(c) from t",
```

Comment:
We should be able to do 3 stage for this kind of case when deliver this feature. It can be done in this story or next task.

Reply:
get, sure